From b091fca4fa4e8c4fce8c113969e8b3f39a89f5aa Mon Sep 17 00:00:00 2001
From: Rafael Palomar <rafael.palomar@rr-research.no>
Date: Wed, 2 Oct 2019 20:10:11 +0200
Subject: [PATCH 05/24] COMP: Adding conditional for including python testing
 in QTCore

With the following configuration:

		-DSlicer_SUPERBUILD=OFF
		-DBUILD_TESTING=OFF
		-DSlicer_BUILD_EXTENSIONMANAGER_SUPPORT=OFF
		-DSlicer_BUILD_CLI_SUPPORT=OFF
		-DSlicer_BUILD_CLI=OFF
		-DCMAKE_CXX_STANDARD=11
		-DSlicer_REQUIRED_QT_VERSION=5
		-DSlicer_BUILD_DICOM_SUPPORT=OFF
		-DSlicer_BUILD_ITKPython=OFF
		-DSlicer_BUILD_QTLOADABLEMODULES=OFF
		-DSlicer_BUILD_QT_DESIGNER_PLUGINS=OFF
		-DSlicer_USE_CTKAPPLAUNCHER=OFF
		-DSlicer_USE_PYTHONQT=OFF
		-DSlicer_USE_QtTesting=OFF
		-DSlicer_USE_SimpleITK=OFF
		-DSlicer_VTK_RENDERING_BACKEND=OpenGL2
		-DSlicer_VTK_VERSION_MAJOR=8
		-DSlicer_INSTALL_DEVELOPMENT=ON
		-DCMAKE_INSTALL_RPATH=/usr/lib64/Slicer-4.11:/usr/lib64/ctk-0.1:/usr/lib64/Slicer-4.11/qt-loadable-modules:/usr/lib64/ITK-5.1.0
		-DCMAKE_BUILD_WITH_INSTALL_RPATH=ON
		-DSlicer_USE_SYSTEM_LibArchive=ON
		-DTeem_DIR=/usr/lib64
		-DjqPlot_DIR=/usr/share/jqPlot

the following error raises:

CMake Error at Base/QTCore/Testing/Python/cmake_install.cmake:41 (file):
  file INSTALL cannot find
  "/var/tmp/portage/sci-medical/3DSlicer-9999/work/3DSlicer-9999_build/Base/QTCore/Testing/Python/qSlicerModuleGenericTest.py.in".
Call Stack (most recent call first):
  Base/QTCore/Testing/cmake_install.cmake:43 (include)
  Base/QTCore/cmake_install.cmake:88 (include)
  Base/cmake_install.cmake:43 (include)
  cmake_install.cmake:165 (include)

This can be solved by introducing a condition for adding the Python
subdirectory in QTCore/Tesitng.
---
 Base/QTCore/Testing/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Base/QTCore/Testing/CMakeLists.txt b/Base/QTCore/Testing/CMakeLists.txt
index 1c97afbfe..e39a6bcf6 100644
--- a/Base/QTCore/Testing/CMakeLists.txt
+++ b/Base/QTCore/Testing/CMakeLists.txt
@@ -1,2 +1,5 @@
 add_subdirectory(Cxx)
-add_subdirectory(Python)
+
+if(Slicer_USE_PYTHONQT)
+  add_subdirectory(Python)
+endif()
-- 
2.24.1

